app/src/main/java/org/oppia/android/app/customview/LessonThumbnailImageView.kt

```diff
@@ -77,26 +83,31 @@ class LessonThumbnailImageView @JvmOverloads constructor(
   }
 
   private fun loadLessonThumbnail() {
-    var transformations = if (lessonThumbnail.isBlurred) {
-      arrayOf(BlurTransformation(context))
+    var transformations = if (isBlurred) {
+      listOf(ImageTransformation.BLUR)
     } else {
-      arrayOf()
+      listOf()
     }
     if (lessonThumbnail.thumbnailFilename.isNotEmpty()) {
-      loadImage(lessonThumbnail.thumbnailFilename, *transformations)
+      loadImage(lessonThumbnail.thumbnailFilename, transformations)
     } else {
-      imageView.setImageResource(getLessonDrawableResource(lessonThumbnail))
+      val glideTransformations: Array<Transformation<Bitmap>> = transformations.map {
+        when (it) {
+          ImageTransformation.BLUR -> BlurTransformation(context)
+          else -> throw IllegalArgumentException("Invalid transformation")
+        }
+      }.toTypedArray()
       Glide.with(context)
         .asBitmap()
         .load(getLessonDrawableResource(lessonThumbnail))
-        .transform(*transformations)
+        .transform(*glideTransformations)
         .into(imageView)
```

Comment:
To keep the transformation code in one place, might it be possible to create a new ImageLoader function which loads a new image based on a drawable?

Reply:
Done. Looks much better, thanks!